File: projects/proj08/ruby/tempTester.rb
# TempTester.rb test the Temperature module
#
# By: Aaron Santucci
# Date: April 11, 2017
##########################################################

require '/home/ajs94/CS214/projects/proj08/ruby/Temperature'

def testTemperature
  
  # test 1
  baseTemp = Temperature::Temperature.new(0, 'F')
  limitTemp = Temperature::Temperature.new(100, 'C')
  stepValue = 2.5
  
  while baseTemp.lessThan(limitTemp)
    puts("Fahrenheit #{baseTemp.getFahren}\t\tCelcius #{baseTemp.getCelcius}\t\tKelvin #{baseTemp.getKelvin}")
    baseTemp.raiseDegrees(stepValue)
  end
  
  print "\n"
  
  # test 2
  baseTemp.degrees = 0
  baseTemp.scale = 'K'
  limitTemp.degrees = 212
  limitTemp.scale = 'F'
  stepValue = 20.0
  
  while baseTemp.lessThan(limitTemp)
    puts("Fahrenheit #{baseTemp.getFahren}\t\tCelcius #{baseTemp.getCelcius}\t\tKelvin #{baseTemp.getKelvin}")
    baseTemp.raiseDegrees(stepValue)
  end
  
  print "All tests passed!\n"
end

testTemperature

File: projects/proj08/java/TempTester.java
/* TempTester.java is the temperature class tester program
 * Student: Aaron Santucci
 * Date: April 10, 2017
 ****************************************************************************/


public class TempTester
{
	public static void main (String [] args)
	{
		Temperature baseTemp = new Temperature(0.0, 'F');
		Temperature limitTemp = new Temperature(100.0, 'C');
		double stepValue = 2.5;
	
		System.out.println("\n");
		
		while (baseTemp.lessThan(limitTemp))
		{
			System.out.println("Fahrenheit " + baseTemp.getFahren() + "\t" + "Celcius " + baseTemp.getCelcius() + "\t" + "Kelvin " + baseTemp.getKelvin());
			baseTemp.raise(stepValue);
		}
		
		System.out.println("\n");
		
		baseTemp = new Temperature(0.0, 'K');
		limitTemp = new Temperature(212.0, 'F');
		stepValue = 20.0;
		
		while (baseTemp.lessThan(limitTemp))
		{
			System.out.println("Fahrenheit " + baseTemp.getFahren() + "\t" + "Celcius " + baseTemp.getCelcius() + "\t" + "Kelvin " + baseTemp.getKelvin());
			baseTemp.raise(stepValue);
		}
	}
	
	
}